assayist/processor/loose_rpm_analyzer.py

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+import glob
+import os
+
+from assayist.common.models import content
+from assayist.processor.base import Analyzer
+from assayist.processor.logging import log
+
+
+class LooseRpmAnalyzer(Analyzer):
+    """Analyze RPMs embedded in the build artifacts."""
+
+    def run(self):
+        """Start the loose RPM analyzer."""
+        build_info = self.read_metadata_file(self.BUILD_FILE)
+        build_id = build_info['id']
+
+        # Remember current directory
+        orig_dir = os.getcwd()
+
+        # Switch to dir of all unpacked content to search for RPM files
+        unpacked_content_path = os.path.join(self.input_dir, self.UNPACKED_ARCHIVES_DIR)
+        os.chdir(unpacked_content_path)
+
+        for archive_type in os.listdir():  # 'rpm', 'container_layer', 'maven', etc.
+            for archive in os.listdir(archive_type):  # dir name == archive filename
+                search_path = os.path.join(archive_type, archive, '**/*.rpm')
+                for loose_rpm in glob.iglob(search_path, recursive=True):
+
+                    loose_rpm_build, loose_rpm_info = self._get_associated_build(loose_rpm)
+                    if not loose_rpm_build:
+                        log.error(f'Cannot find build for {loose_rpm} embedded in {archive} in '
+                                  f'build {build_id}')
```

Comment:
Should a `continue` be added after this log statement? If `loose_rpm_build` is `None`, then `loose_rpm_info` will be `None`, and it's being used below.

Reply:
Nice catch, added!